### PR TITLE
Step 4: Add a page for user to make commets by using HTML form tag.

### PR DESCRIPTION
**Pull request by Phoebe-Khokgawe**

### CODE REVIEW

File: portfolio/pom.xml [hunk 1]
```diff
@@ -41,7 +41,7 @@
         <version>2.2.0</version>
         <configuration>
           <!-- TODO: set project ID. -->
-          <deploy.projectId>phoebek-step-2020</deploy.projectId>
+          <deploy.projectId>YOUR_PROJECT_ID_HERE</deploy.projectId>
```
Comment (leonnemets) [inline portfolio/pom.xml]: Seems accidental. Did you mean to do this?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -43,8 +43,36 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+    response.setContentType("text/html;");
+    response.getWriter().println("Thanks for your comment: \n\"" + text + "\"\n Hope you have a good day!");
+    
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (leonnemets) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:58]: Can this be a static function?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -43,8 +43,36 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
```
Comment (leonnemets) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: return value == null ? defaultValue : value;

File: portfolio/pom.xml (hunk 1 above)
Comment (Phoebe-Khokgawe) [reply portfolio/pom.xml]: I thought that was causing a bug but I changed it back now in the later branches.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (Phoebe-Khokgawe) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:58]: I'm not sure sorry.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (Phoebe-Khokgawe) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I improved this function and push it to  b68273de19ff9722a05935adf40f01ea1f1a3652 this commit https://github.com/Phoebe-Khokgawe/STEP_Starter/commit/b68273de19ff9722a05935adf40f01ea1f1a3652#diff-23a0cb4c16cf41219c2807c8ac14ee1d7c554d240941f51ba63f10cc057b10ba.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (leonnemets) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:58]: this function does not use any state from this class and can be made static. https://www.edureka.co/blog/java-static-method/

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (leonnemets) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nice

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -27,9 +27,10 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    int randomIndex = (int)Math.floor(Math.random() * ((int)jsonString.size()));
-    System.out.print(jsonString.size() + " randome num: " + randomIndex);
-    response.getWriter().println(jsonString.get(randomIndex).toString());
+    String jsonString = "{ \"Animal\" : \"CAT\" }";
+    // int randomIndex = (int)Math.floor(Math.random() * ((int)jsonString.size()));
+    // System.out.print(jsonString.size() + " randome num: " + randomIndex);
```
Comment (leonnemets) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:32]: Any reason to keep this around?

**Review by leonnemets — APPROVED** [review]

**Pull request merged**